openff/toolkit/topology/topology.py

```diff
@@ -1175,9 +1175,13 @@ def _identify_chemically_identical_molecules(
                 if mol2_idx in already_matched_mols:
                     continue
                 mol2 = self.molecule(mol2_idx)
-                are_isomorphic, atom_map = Molecule.are_isomorphic(
-                    mol1, mol2, return_atom_map=True
-                )
+                if isinstance(mol1, type(mol2)) and isinstance(mol2, type(mol1)):
```

Comment:
This turned out uglier than I thought it would be ... I think this basically just checks if they're the same class since I'm not sure there's any other way two classes can "be" subclasses of each other. I guess just having it try this logic if one is a subclass of the other (but not necessarily both) might handle what you're after:


```suggestion
                if isinstance(mol1, type(mol2)) or isinstance(mol2, type(mol1)):
```

This should catch the case of comparing `Molecule` to somebody else's custom class that subclasses out of `FrozenMolecule`

Reply:
Good call, I like the "or". Thanks for the nice explanation.